OpenKeychain/src/main/java/org/sufficientlysecure/keychain/pgp/CanonicalizedSecretKey.java

```diff
@@ -252,6 +253,49 @@ public PGPSignatureGenerator getCertSignatureGenerator(Map<ByteBuffer, byte[]> s
         }
     }
 
+    public PGPAuthSignatureGenerator getAuthSignatureGenerator(int hashAlgo, boolean cleartext,
+                                                               Map<ByteBuffer, byte[]> signedHashes, Date creationTimestamp)
+            throws PgpGeneralException {
+        if (mPrivateKeyState == PRIVATE_KEY_STATE_LOCKED) {
+            throw new PrivateKeyNotUnlockedException();
+        }
+
+        // We explicitly create a signature creation timestamp in this place.
+        // That way, we can inject an artificial one from outside, ie the one
+        // used in previous runs of this function.
+        if (creationTimestamp == null) {
+            // to sign using nfc PgpSignEncrypt is executed two times.
+            // the first time it stops to return the PendingIntent for nfc connection and signing the hash
+            // the second time the signed hash is used.
+            // to get the same hash we cache the timestamp for the second round!
+            creationTimestamp = new Date();
+        }
```

Comment:
As far as I understand: You do not use creation time for authentication, thus we can get rid of ``creationTimestamp`` parameter and this whole block.

Reply:
Yes, thanks. I have now removed unecessary timestamp parameters in other places too.